### PR TITLE
[1.8.9] Entity Mappings

### PR DESCRIPTION
**Pull request by Grayray75**
Most of the mappings are taken from 1.16.5.

Also:
* Fixes publish workflow.
* Updates Gradle, Enigma and more.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-bin.zip
```
Comment (BluCobalt) [inline gradle/wrapper/gradle-wrapper.properties:3]: I don't see a need for gradle 7, why not 6.8.2?

**Review by BluCobalt — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Grayray75) [reply gradle/wrapper/gradle-wrapper.properties:3]: So that I can use java 16

Comment (BluCobalt) [reply gradle/wrapper/gradle-wrapper.properties:3]: i see

**Review by BoogieMonster1O1 — APPROVED** [review]

**Comment by BoogieMonster1O1** [comment]
oops

**Pull request merged**